go.mod

```diff
@@ -241,12 +236,79 @@ require (
 	gopkg.in/inf.v0 v0.9.1 // indirect
 	gopkg.in/warnings.v0 v0.1.2 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
-	k8s.io/kube-openapi v0.0.0-20241212222426-2c72e554b1e7 // indirect
-	k8s.io/kubectl v0.32.1 // indirect
+	k8s.io/api v0.32.2
+	k8s.io/apiextensions-apiserver v0.32.2
+	k8s.io/apimachinery v0.32.2
+	k8s.io/apiserver v0.32.2
+	k8s.io/cli-runtime v0.32.2
+	k8s.io/client-go v0.32.2
+	k8s.io/component-base v0.32.2
+	k8s.io/component-helpers v0.32.2 // indirect
+	k8s.io/controller-manager v0.32.2 // indirect
+	k8s.io/kubectl v0.32.2 // indirect
+	k8s.io/kubernetes v1.32.2
 	oras.land/oras-go v1.2.5 // indirect
 	sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.31.0 // indirect
 	sigs.k8s.io/json v0.0.0-20241014173422-cfa47c3a1cc8 // indirect
 	sigs.k8s.io/kustomize/api v0.18.0 // indirect
 	sigs.k8s.io/kustomize/kyaml v0.18.1 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.5.0 // indirect
 )
+
+replace k8s.io/api => k8s.io/api v0.32.2
```

Comment:
Why do we need to update the k8s api and add the replaces?
When we update the controller runtime, we will have all updates right.

Reply:
These replace are added by the tool that is new with this PR: `hack/tools/k8sMaintainer.go`. There is also an explainer on that code at `hack/tools/k8sMaintainer_doc.md` (which maybe should go away or be moved, but is perhaps helpful for at least this review). TL;DR; the replace keep us from importing the whole kitchen sink of `k8s.io/kubernetes`, pinning on the libraries that otherwise would end up as `requires` on staging versions at v0.0.0

Reply:
Discussed in the slack I think we are all agreed that we will add the replacements so closing it as sorted out